run-windows broken on versions < 0.70 when VS2022 is installed

### Problem Description

RNW versions < 0.70 don't build when using VS2022.

```shell
X:\rnw69\node_modules\react-native-windows\Microsoft.ReactNative\Utils\Helpers.h(21,8): error C2187: syntax error: 'typename' was unexpected here (compiling source file Modules\AppStateModule.cpp) [X:\
       rnw69\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj]
```

### Steps To Reproduce

1. Install Visual Studio 2022
2. npx react-native init rnw69 --template react-native@^0.69
3. cd rnw69
4. npx react-native-windows-init --overwrite
5. npx react-native run-windows

### Expected Results

_No response_

### CLI version

8.0.6

### Environment

```markdown
info Fetching system and libraries information...
System:
    OS: Windows 10 10.0.25246
    CPU: (8) x64 Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
    Memory: 39.58 GB / 63.85 GB
  Binaries:
    Node: 16.15.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 8.11.0 - C:\Program Files\nodejs\npm.CMD
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
    Windows SDK:
      AllowDevelopmentWithoutDevLicense: Enabled
      AllowAllTrustedApps: Enabled
      Versions: 10.0.18362.0, 10.0.19041.0, 10.0.22000.0, 10.0.22621.0
  IDEs:
    Android Studio: AI-213.7172.25.2113.9014738
    Visual Studio: 17.4.33110.190 (Visual Studio Enterprise 2022), 16.11.33027.164 (Visual Studio Enterprise 2019)
  Languages:
    Java: 11.0.16.1 - C:\Program Files\Eclipse Adoptium\jdk-11.0.16.101-hotspot\bin\javac.EXE
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0
    react-native: 0.69.7 => 0.69.7
    react-native-windows: 0.69.13 => 0.69.13
  npmGlobalPackages:
    *react-native*: Not Found
```


### Target Platform Version

_No response_

### Target Device(s)

_No response_

### Visual Studio Version

_No response_

### Build Configuration

_No response_

### Snack, code example, screenshot, or link to a repository

![image](https://user-images.githubusercontent.com/1422161/203197936-93b97af4-13c6-41b9-a9f0-8413bc94c72d.png)

--- a/change/@react-native-windows-cli-8fb1e274-a5a2-4a7c-9dae-2ed655b8f87e.json
+++ b/change/@react-native-windows-cli-8fb1e274-a5a2-4a7c-9dae-2ed655b8f87e.json
@@ -1,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Fix CLI to search for correct VS version range",
+  "packageName": "@react-native-windows/cli",
+  "email": "jthysell@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/packages/@react-native-windows/cli/src/runWindows/utils/vsInstalls.ts
+++ b/packages/@react-native-windows/cli/src/runWindows/utils/vsInstalls.ts
@@ -8,6 +8,7 @@ import {CodedError} from '@react-native-windows/telemetry';
 import {execSync} from 'child_process';
 import fs from '@react-native-windows/fs';
 import path from 'path';
+import semver from 'semver';
 
 /**
  * A subset of the per-instance properties returned by vswhere
@@ -61,15 +62,46 @@ function vsWhere(args: string[], verbose?: boolean): any[] {
  */
 export function enumerateVsInstalls(opts: {
   requires?: string[];
-  version?: string;
+  minVersion?: string;
   verbose?: boolean;
   latest?: boolean;
   prerelease?: boolean;
 }): VisualStudioInstallation[] {
   const args: string[] = [];
 
-  if (opts.version) {
-    args.push(`-version [${opts.version},${Number(opts.version) + 1})`);
+  if (opts.minVersion) {
+    // VS 2019 ex: minVersion == 16.7 => [16.7,17.0)
+    // VS 2022 ex: minVersion == 17.0 => [17.0,18.0)
+
+    // Try to parse minVersion as both a Number and SemVer
+    const minVersionNum = Number(opts.minVersion);
+    const minVersionSemVer = semver.parse(opts.minVersion);
+
+    let minVersion: string;
+    let maxVersion: string;
+
+    if (minVersionSemVer) {
+      minVersion = minVersionSemVer.toString();
+      maxVersion = (minVersionSemVer.major + 1).toFixed(1);
+    } else if (!Number.isNaN(minVersionNum)) {
+      minVersion = minVersionNum.toFixed(1);
+      maxVersion = (Math.floor(minVersionNum) + 1).toFixed(1);
+    } else {
+      // Unable to parse minVersion and determine maxVersion,
+      // caller will throw error that version couldn't be found.
+      return [];
+    }
+
+    const versionRange =
+      `[${minVersion},${maxVersion}` + (opts.prerelease ? ']' : ')');
+
+    if (opts.verbose) {
+      console.log(
+        `Looking for VS installs with version range: ${versionRange}`,
+      );
+    }
+
+    args.push(`-version ${versionRange}`);
   }
 
   if (opts.requires) {

